concClient.go

```diff
@@ -0,0 +1,385 @@
+package aptos
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/aptos-labs/aptos-go-sdk/api"
+	"github.com/hasura/go-graphql-client"
+	"io"
+	"net/url"
+	"sort"
+	"strconv"
+	"time"
+)
+
+// ConcClient is a concurrent client using Go routines, to wrap the existing [Client].  This is a testing ground, and the future go routine APIs will hopefully be merged directly into the [Client].
+type ConcClient struct {
+	client *Client
+}
+
+// ConcResponse is a concurrent response wrapper as a return type for all APIs.  It is meant to specifically be used in channels.
+type ConcResponse[T any] struct {
+	Result T
+	Err    error
+}
+
+// NewConcClient creates a [ConcClient]
+func NewConcClient(c *Client) (*ConcClient, error) {
+	if c == nil {
+		return nil, errors.New("nil client")
+	}
+	return &ConcClient{client: c}, nil
+}
+
+// SetTimeout adjusts the HTTP client timeout
+//
+//	client.SetTimeout(5 * time.Millisecond)
+func (cc *ConcClient) SetTimeout(timeout time.Duration) {
+	cc.client.nodeClient.client.Timeout = timeout
+}
+
+// TODO: Support multiple output channels?
```

Comment:
no need? multiple workers can read from a channel if needed. (not like Rust mpsc 'multiple producer single consumer', the default Go chan is good for multiple-producer and multiple consumer)